### ISSUE TITLE
SwipeDown on modal doesn't close anymore in v6

### ISSUE THREAD

**Issue by ptgamr** [react-navigation/react-navigation#10856]
### Current behavior

Swipe down on modal, modal re-popup again.

Re-producible on the examples `ModalStack` inside this repo, just adding `gestureEnabled = true` to `screenOptions`
https://github.com/ptgamr/react-navigation/blob/main/example/src/Screens/ModalStack.tsx#L86

A bit of digging shows me this line is the culprit: https://github.com/react-navigation/react-navigation/blob/main/packages/stack/src/views/Stack/Card.tsx#L147

calling `animate` there will cancel the `setTimeout` that call `onClose()`

### Expected behavior

Modal should be close

### Reproduction

https://github.com/ptgamr/react-navigation

### Platform

- [X] Android
- [X] iOS
- [ ] Web
- [ ] Windows
- [ ] MacOS

### Packages

- [ ] @react-navigation/​bottom-tabs
- [ ] @react-navigation/​drawer
- [ ] @react-navigation/​material-bottom-tabs
- [ ] @react-navigation/​material-top-tabs
- [X] @react-navigation/​stack
- [ ] @react-navigation/​native-stack

### Environment

- [x] I've removed the packages that I don't use

| package                                | version |
| -------------------------------------- | ------- |
| @react-navigation/native               |  ^6.0.13
| @react-navigation/bottom-tabs          | ^6.4.0
| @react-navigation/drawer               | ^6.5.0
| @react-navigation/stack                | ^6.3.0
| react-native-safe-area-context         | ^4.3.3
| react-native-screens                   | ^3.17.0
| react-native-gesture-handler           | ^2.6.1
| react-native-reanimated                | ^2.10.0
| react-native                           |  0.69.3

**Comment by EgorFrom** [comment]
Don't know all of this, but if anyone stuck on it you can do this:
Before:
```
this.animate({ closing, velocity });

        if (closing) {
          // We call onClose with a delay to make sure that the animation has already started
          // This will make sure that the state update caused by this doesn't affect start of animation
          this.pendingGestureCallback = setTimeout(() => {
            onClose();

            // Trigger an update after we dispatch the action to remove the screen
            // This will make sure that we check if the screen didn't get removed so we can cancel the animation
            this.forceUpdate();
          }, 32) as any as number;
        }
```

You can do:
```
if (closing) {
          // We call onClose with a delay to make sure that the animation has already started
          // This will make sure that the state update caused by this doesn't affect start of animation
          this.pendingGestureCallback = setTimeout(() => {
            onClose();

            // Trigger an update after we dispatch the action to remove the screen
            // This will make sure that we check if the screen didn't get removed so we can cancel the animation
            this.forceUpdate();
          }, 32) as any as number;
        } else {
          this.animate({ closing, velocity });
        }
```

**Comment by jacksonHendric** [comment]
I have the same issue, it's breaking a lot of functionality right now. Any updates on when this will be fixed? If not what can we downgrade to?
This is on `stack` not `native-stack`

**Comment by ptgamr** [comment]
@EgorFrom yeah, that was better than my fix. I commented out the `this.animate()` inside `componentDidUpdate` and other thing breaks like `navigation.goBack()` no longer close the modal.

**Comment by ptgamr** [comment]
Actually, using @EgorFrom solution doesn't really fix it for me, it make the modal doesn't completely go away sometimes.

I am currently using this fix:

```
diff --git a/node_modules/@react-navigation/stack/src/views/Stack/Card.tsx b/node_modules/@react-navigation/stack/src/views/Stack/Card.tsx
index 820969c..726a08f 100755
--- a/node_modules/@react-navigation/stack/src/views/Stack/Card.tsx
+++ b/node_modules/@react-navigation/stack/src/views/Stack/Card.tsx
@@ -144,7 +144,13 @@ export default class Card extends React.Component<Props, State> {
       // When route was closed due to a gesture, the animation would've happened already
       // It's still important to trigger the animation so that `onClose` is called
       // If `onClose` is not called, cleanup step won't be performed for gestures
-      this.animate({ closing });
+
+      // Temp fix for this issue: https://github.com/react-navigation/react-navigation/issues/10856
+      // if there is a pendingGestureCallback, we should not call animate
+      // otherwise, it will clear out that callback
+      if (!this.pendingGestureCallback) {
+        this.animate({ closing });
+      }
     }
   }
```

**Comment by jacksonHendric** [comment]
@ptgamr This is the fix that seems to work. Anyone working on a PR for this?

**Comment by enchorb** [comment]
Have this issue after upgrading to `@react-navigation/stack@6.3.0` was fine in the previous version

**Comment by ptgamr** [comment]
@jacksonHendric not yet, as I'm not even sure this is the root cause - maybe there is a deeper reason for this ... would love to have some contributor's eyes on this.

**Comment by ptgamr** [comment]
@enchorb use `patch-package` and apply the fix above seems to fix it for me

**Comment by espenjanson** [comment]
+1 on this bug, +1 @ptgamr patch worked for me

**Comment by kacperkapusciak** [comment]
Thanks for submitting an issue!

Released `@react-navigation/stack@6.3.1` that reverts a faulty commit that introduced this bug.

Cheers
